src/component-library/pages/Rule/list/ListRule.tsx

```diff
@@ -1,33 +1,282 @@
+    };
+
+    return (
+        <div className="flex space-x-2">
+            <Input className="w-full sm:flex-grow" onChange={handleInputChange} onEnterKey={onSearch} placeholder="Current Account" value={value} />
+            <FilterDropdownButton isExpanded={isFilterExpanded} onToggle={onFilterToggle} />
+        </div>
+    );
+};
+
+function FilterField({ children, label }: { children: React.ReactNode; label: string }) {
+    return (
+        <div className="grow flex-1">
+            <div className="text-neutral-900 dark:text-neutral-100 mb-2">{label}</div>
+            {children}
+        </div>
+    );
+}
+
+const FilterInputs = ({ filters, onFiltersChange }: FilterInputsProps) => {
+    const onActivityChange = (event: ChangeEvent<HTMLInputElement>) => {
+        onFiltersChange({ activity: event.target.value });
+    };
+
+    const onStateChange = (value: string) => {
+        if (value === RuleState.UNKNOWN) {
+            onFiltersChange({ state: undefined });
+        } else {
+            onFiltersChange({ state: value as RuleState });
+        }
+    };
+
+    const onScopeChange = (event: ChangeEvent<HTMLInputElement>) => {
+        const newScope = event.target.value || DEFAULT_SCOPE;
+        onFiltersChange({ scope: newScope });
+    };
+
+    const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+        onFiltersChange({ name: event.target.value });
+    };
+
+    const formatDate = (date?: Date): string => {
+        if (!date) return '';
+        return date.toISOString().split('T')[0];
+    };
+
+    const onDateChange = (event: FormEvent<HTMLInputElement>, dateType: 'updatedAfter' | 'updatedBefore') => {
+        const date = new Date(event.currentTarget.value);
+        if (isNaN(date.getTime())) {
+            onFiltersChange({ [dateType]: undefined });
+            return;
+        }
+        onFiltersChange({ [dateType]: date });
+    };
+
+    const rootClass = 'flex flex-col sm:flex-row sm:items-center w-full space-y-2 sm:space-y-0 sm:space-x-2';
+
+    return (
+        <>
+            <div className={rootClass}>
+                <FilterField label="Activity">
+                    <Input className="w-full" value={filters.activity} onChange={onActivityChange} placeholder="Any Activity" />
+                </FilterField>
+                <FilterField label="State">
+                    <Select onValueChange={onStateChange} defaultValue={undefined} value={filters.state}>
+                        <SelectTrigger className="w-full">
+                            <SelectValue placeholder="Any" />
+                        </SelectTrigger>
+                        <SelectContent>
+                            <SelectItem value={RuleState.UNKNOWN}>Any</SelectItem>
+                            {Object.values(RuleState).map(state => {
+                                if (state === RuleState.UNKNOWN) return null;
+                                // TODO: transform RuleState to a more user-friendly string
```

Comment:
This can be really quickly addressed after #610 is merged, as there are utilities for this transformation. Waiting_Approval should be displayed without the underscore (that's the only problem currently).